### DISCUSSION TITLE
implicit constructor to func conversion

### DISCUSSION THREAD

**Discussion by DanielIndie95** [dotnet/csharplang#885] [General]
Wherever the parameter required in a function is a Func<T, T2> you can supply a T2 constructor that requires T as a parameter

for example you can do something like that 
 ```csharp
 new []{5}.Select(new List<object>);
 ```
 which is the equivalent to this :
 ```csharp
 new []{5}.Select(num => new List<object>(num));
```
i think it looks cleaner

(i opened a pull-request before i knew i needed to create issue first #884 )

**Comment by orthoxerox** [comment]
See also #84, #598

**Comment by chaosrealm** [comment]
The fact that constructors cannot be used as functions has been a PITA in many situations. Would love to see this added to the language.

**Comment by CyrusNajmabadi** [comment]
```new List<object>``` seems way too subtle.  How about ```List<object>.new``` instead.

**Comment by jnm2** [comment]
I've preferred `List<object>.new`.

**Comment by yaakov-h** [comment]
I like it, but what if there's already a static member named `new`? 😈

**Comment by svick** [comment]
@yaakov-h C# considers `new` to be a keyword. If there really was a member called `new`, you would have to refer to it as `@new` from C# already.

**Comment by orthoxerox** [comment]
Is it not a reserved keyword?

**Comment by yaakov-h** [comment]
Good point, nvm.
